app/src/main/java/com/example/topmovies/viewmodel/MovieViewModel.kt

```diff
@@ -1,61 +1,60 @@
 package com.example.topmovies.viewmodel
 
 import android.content.SharedPreferences
+import android.os.Handler
+import android.os.Looper
 import androidx.lifecycle.LiveData
 import androidx.lifecycle.MutableLiveData
-import com.example.topmovies.model.Movie
+import com.example.topmovies.model.MovieEntity
 import com.example.topmovies.repository.MovieRepository
 import com.example.topmovies.unit.EnumScreen
 
 class MovieViewModel constructor(
     private val repository: MovieRepository,
-    private val favoritePref: SharedPreferences,
-    sharedPref: SharedPreferences
+    sharedPref: SharedPreferences,
 ) : BaseViewModel(sharedPref) {
 
-    private val movies = MutableLiveData<List<Movie>>()
-    private val _favoriteMovies = mutableListOf<Movie>()
-    private val favoriteMovies: MutableLiveData<List<Movie>> = MutableLiveData(_favoriteMovies)
+    private val movies = MutableLiveData<List<MovieEntity>>()
+    private val favoriteMovies = MutableLiveData<List<MovieEntity>>()
     private var _errorMessage: String? = null
     val errorMessage = MutableLiveData(_errorMessage)
+    private val handler = Handler(Looper.getMainLooper())
 
-    fun getMoviesList(screen: EnumScreen): LiveData<List<Movie>> {
+    fun getMoviesList(screen: EnumScreen): LiveData<List<MovieEntity>> {
         return when (screen) {
             EnumScreen.MOVIES -> movies
             EnumScreen.FAVORITE -> favoriteMovies
         }
     }
 
     fun getMovies() {
```

Comment:
What if I don't want to show anything before all data is loaded?
If your DB is empty you might end up in an inconsistent state.
For favorite movies, you aren't taking the empty DB condition into consideration.